src/main/java/appeng/me/cache/EnergyGridCache.java

```diff
@@ -468,8 +459,32 @@ public boolean isNetworkPowered()
 	@Override
 	public double injectPower( final double amt, final Actionable mode )
 	{
-		this.localSeen.clear();
-		return this.injectAEPower( amt, mode, this.localSeen );
+		final Queue<IEnergyGridProvider> toVisit = new PriorityQueue<>(COMPARATOR_LOWEST_PERCENTAGE_FIRST);
+		final Set<IEnergyGridProvider> visited = new HashSet<>();
+		toVisit.add( this );
+
+		double leftover = amt;
+
+		while( !toVisit.isEmpty() && leftover > 0 )
+		{
+			final IEnergyGridProvider next = toVisit.poll();
+			visited.add( next );
+
+			final double cannotHold = next.injectProviderPower( amt, Actionable.SIMULATE, visited );
+			leftover = next.injectProviderPower( leftover - cannotHold, mode, visited );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "leftover". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1854)